app/src/main/java/com/eslamwaheed/mvvmdemo/data/AppDatabase.kt

```diff
@@ -0,0 +1,33 @@
+package com.eslamwaheed.mvvmdemo.data
+
+import android.content.Context
+import androidx.room.Database
+import androidx.room.Room
+import androidx.room.RoomDatabase
+import com.eslamwaheed.mvvmdemo.data.daos.PhotosDAO
+import com.eslamwaheed.mvvmdemo.data.models.UnsplashSearchResponse
+
+@Database(
+    entities = [UnsplashSearchResponse.Result::class, UnsplashSearchResponse.Result.Urls::class],
+    version = 1,
+    exportSchema = false
+)
+abstract class AppDatabase : RoomDatabase() {
+    abstract fun photosDao(): PhotosDAO
+
+    companion object {
+        @Volatile
```

Comment:
using @Volatile and synchronized is very good for thread-safe but we need to use dependency injection here

Reply:
done